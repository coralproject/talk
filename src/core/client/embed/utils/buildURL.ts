export default function buildURL({
  protocol = window.location.protocol,
  hostname = window.location.hostname,
  port = window.location.port,
  pathname = window.location.pathname,
  search = window.location.search,
  hash = window.location.hash,
} = {}) {
  if (search && search[0] !== "?") {
    search = `?${search}`;
  } else if (search === "?") {
    search = "";
  }
  return `${protocol}//${hostname}${
    port ? `:${port}` : ""
  }${pathname}${search}${hash}`;
}
