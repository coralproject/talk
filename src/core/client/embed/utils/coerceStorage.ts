import { createInMemoryStorage } from "coral-framework/lib/storage";

function coerceStorage(type: "localStorage" | "sessionStorage"): Storage {
  let storage: Storage;
  try {
    storage = window[type];
  } catch {
    // eslint-disable-next-line no-console
    console.debug(`${type} not available, use in-memory replacement`);

    storage = createInMemoryStorage();
  }

  return storage;
}

export default coerceStorage;
