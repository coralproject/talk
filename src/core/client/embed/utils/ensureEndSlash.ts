export default function ensureEndSlash(p: string) {
  return p.match(/\/$/) ? p : `${p}/`;
}
