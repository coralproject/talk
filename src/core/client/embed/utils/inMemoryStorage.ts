class InMemoryStorage implements Storage {
  private readonly data = new Map<string, string>();

  public get length() {
    return this.data.size;
  }

  public clear() {
    this.data.clear();
  }

  public key(n: number) {
    const keys = [...this.data.keys()];
    if (keys.length < n || n < 0) {
      return null;
    }

    return keys[n];
  }

  public getItem(key: string): string | null {
    return this.data.get(key) || null;
  }

  public removeItem(key: string): void {
    this.data.delete(key);
  }

  public setItem(key: string, value: string): void {
    this.data.set(key, value);
  }
}

export default InMemoryStorage;
