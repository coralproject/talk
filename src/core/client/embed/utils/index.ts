export { default as buildURL } from "./buildURL";
export { default as ensureEndSlash } from "./ensureEndSlash";
export { default as ensureNoEndSlash } from "./ensureNoEndSlash";
export { default as startsWith } from "./startsWith";
export { default as prefixStorage } from "./prefixStorage";
export { default as parseHashQuery } from "./parseHashQuery";
