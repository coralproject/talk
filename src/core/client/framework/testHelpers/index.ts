export {
  default as createRelayEnvironment,
  CreateRelayEnvironmentParams,
} from "./createRelayEnvironment";
export { default as createFluentBundle } from "./createFluentBundle";
export { default as createSinonStub } from "./createSinonStub";
export {
  default as removeFragmentRefs,
  NoFragmentRefs,
} from "./removeFragmentRefs";
export { default as createUUIDGenerator } from "./createUUIDGenerator";
export * from "./denormalize";
export { default as replaceHistoryLocation } from "./replaceHistoryLocation";
