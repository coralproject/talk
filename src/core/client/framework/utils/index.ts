export { default as buildURL } from "./buildURL";
export { default as parseURL } from "./parseURL";
export { default as modifyQuery } from "./modifyQuery";
