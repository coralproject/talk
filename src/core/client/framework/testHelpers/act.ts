import TestRenderer from "react-test-renderer";

import { isPromiseLike } from "coral-common/utils";

export default function act<T>(callback: () => T): T {
  let callbackResult: T;
  const actResult = TestRenderer.act(() => {
    callbackResult = callback();
    return callbackResult as any;
  });
  if (isPromiseLike(callbackResult!)) {
    // Return it this way, to preserve warnings that React emits.
    return {
      then(resolve: (value: any) => void, reject: (err?: Error) => void) {
        (actResult as PromiseLike<any>).then(() => {
          resolve(callbackResult);
        }, reject);
      },
    } as any;
  }
  return callbackResult!;
}
