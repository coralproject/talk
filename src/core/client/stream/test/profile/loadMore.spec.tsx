import { ReactTestRenderer } from "react-test-renderer";
import sinon from "sinon";

import { timeout } from "talk-common/utils";
import { createSinonStub } from "talk-framework/testHelpers";

import { assets, comments, meWithComments } from "../fixtures";
import create from "./create";

let testRenderer: ReactTestRenderer;
beforeEach(() => {
  const meStub = {
    ...meWithComments,
    comments: createSinonStub(
      s => s.throws(),
      s =>
        s.withArgs({ first: 5, orderBy: "CREATED_AT_DESC" }).returns({
          edges: [
            {
              node: { ...comments[0], asset: assets[0] },
              cursor: comments[0].createdAt,
            },
            {
              node: { ...comments[1], asset: assets[0] },
              cursor: comments[1].createdAt,
            },
          ],
          pageInfo: {
            endCursor: comments[1].createdAt,
            hasNextPage: true,
          },
        }),
      s =>
        s
          .withArgs({
            first: 10,
            orderBy: "CREATED_AT_DESC",
            after: comments[1].createdAt,
          })
          .returns({
            edges: [
              {
                node: { ...comments[2], asset: assets[0] },
                cursor: comments[2].createdAt,
              },
            ],
            pageInfo: {
              endCursor: comments[2].createdAt,
              hasNextPage: false,
            },
          })
    ),
  };

  const resolvers = {
    Query: {
      asset: createSinonStub(
        s => s.throws(),
        s =>
          s
            .withArgs(undefined, { id: assets[0].id, url: null })
            .returns(assets[0])
      ),
      me: sinon.stub().returns(meStub),
    },
  };

  ({ testRenderer } = create({
    // Set this to true, to see graphql responses.
    logNetwork: false,
    resolvers,
    initLocalState: localRecord => {
      localRecord.setValue(assets[0].id, "assetID");
    },
  }));
});

it("renders comment stream", async () => {
  // Wait for loading.
  await timeout();
  expect(testRenderer.toJSON()).toMatchSnapshot();
});

it("loads more comments", async () => {
  testRenderer.root
    .findByProps({ id: "talk-profile-commentHistory-loadMore" })
    .props.onClick();

  // Wait for loading.
  await timeout();
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
