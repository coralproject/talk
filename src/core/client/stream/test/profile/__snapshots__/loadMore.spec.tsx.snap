// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads more comments 1`] = `
<div
  className="HorizontalGutter-root App-root HorizontalGutter-full"
>
  <ul
    className="TabBar-root TabBar-primary"
    role="tablist"
  >
    <li
      className="Tab-root"
      id="tab-COMMENTS"
      role="presentation"
    >
      <button
        aria-controls="tabPane-COMMENTS"
        aria-selected={false}
        className="BaseButton-root Tab-button Tab-primary"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        role="tab"
        type="button"
      >
        <span>
          ⁨2⁩ Comments
        </span>
      </button>
    </li>
  </ul>
  <section
    aria-labelledby="tab-PROFILE"
    className="App-tabContent"
    id="tabPane-PROFILE"
    role="tabpanel"
  >
    <div
      className="HorizontalGutter-root HorizontalGutter-double"
    >
      <div
        className="Flex-root"
      >
        <div
          className="Flex-flex Flex-halfItemGutter Flex-wrap"
        >
          <div
            className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
          >
            Signed in as 
            <span
              className="Typography-root Typography-bodyCopyBold Typography-colorTextPrimary"
            >
              Markus
            </span>
            .
          </div>
          <div
            className="Flex-root Typography-root Typography-bodyCopy Typography-colorTextPrimary Flex-flex"
          >
            <span>
              Not you? 
            </span>
            <button
              className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              type="button"
            >
              Sign Out
            </button>
          </div>
        </div>
      </div>
      <div
        className="HorizontalGutter-root HorizontalGutter-double"
      >
        <h1
          className="Typography-root Typography-heading3 Typography-colorTextPrimary"
        >
          Comment History
        </h1>
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        >
          <h1
            className="Typography-root Typography-heading4 Typography-colorTextPrimary"
          >
            Story: ⁨title⁩
          </h1>
          <time
            className="Timestamp-root RelativeTime-root"
            dateTime="2018-07-06T18:24:00.000Z"
            title="2018-07-06T18:24:00.000Z"
          >
            2018-07-06T18:24:00.000Z
          </time>
          <div
            className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
          >
            <div
              className="HTMLContent-root"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Joining Too",
                }
              }
            />
          </div>
          <div
            className="Flex-root Flex-flex Flex-itemGutter Flex-alignCenter Flex-directionRow"
          >
            <a
              className="BaseButton-root Button-root Button-sizeRegular Button-colorRegular Button-variantUnderlined"
              href="http://localhost/stories/story-1?commentID=comment-0"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              target="_parent"
              type="button"
            >
              <span
                aria-hidden="true"
                className="Icon-root Icon-sm"
              >
                launch
              </span>
              <span>
                View Conversation
              </span>
            </a>
          </div>
        </div>
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        >
          <h1
            className="Typography-root Typography-heading4 Typography-colorTextPrimary"
          >
            Story: ⁨title⁩
          </h1>
          <time
            className="Timestamp-root RelativeTime-root"
            dateTime="2018-07-06T18:24:00.000Z"
            title="2018-07-06T18:24:00.000Z"
          >
            2018-07-06T18:24:00.000Z
          </time>
          <div
            className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
          >
            <div
              className="HTMLContent-root"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "What's up?",
                }
              }
            />
          </div>
          <div
            className="Flex-root Flex-flex Flex-itemGutter Flex-alignCenter Flex-directionRow"
          >
            <a
              className="BaseButton-root Button-root Button-sizeRegular Button-colorRegular Button-variantUnderlined"
              href="http://localhost/stories/story-1?commentID=comment-1"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              target="_parent"
              type="button"
            >
              <span
                aria-hidden="true"
                className="Icon-root Icon-sm"
              >
                launch
              </span>
              <span>
                View Conversation
              </span>
            </a>
          </div>
        </div>
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        >
          <h1
            className="Typography-root Typography-heading4 Typography-colorTextPrimary"
          >
            Story: ⁨title⁩
          </h1>
          <time
            className="Timestamp-root RelativeTime-root"
            dateTime="2018-07-06T18:24:00.000Z"
            title="2018-07-06T18:24:00.000Z"
          >
            2018-07-06T18:24:00.000Z
          </time>
          <div
            className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
          >
            <div
              className="HTMLContent-root"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Hey!",
                }
              }
            />
          </div>
          <div
            className="Flex-root Flex-flex Flex-itemGutter Flex-alignCenter Flex-directionRow"
          >
            <a
              className="BaseButton-root Button-root Button-sizeRegular Button-colorRegular Button-variantUnderlined"
              href="http://localhost/stories/story-1?commentID=comment-2"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              target="_parent"
              type="button"
            >
              <span
                aria-hidden="true"
                className="Icon-root Icon-sm"
              >
                launch
              </span>
              <span>
                View Conversation
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`renders comment stream 1`] = `
<div
  className="HorizontalGutter-root App-root HorizontalGutter-full"
>
  <ul
    className="TabBar-root TabBar-primary"
    role="tablist"
  >
    <li
      className="Tab-root"
      id="tab-COMMENTS"
      role="presentation"
    >
      <button
        aria-controls="tabPane-COMMENTS"
        aria-selected={false}
        className="BaseButton-root Tab-button Tab-primary"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        role="tab"
        type="button"
      >
        <span>
          ⁨2⁩ Comments
        </span>
      </button>
    </li>
  </ul>
  <section
    aria-labelledby="tab-PROFILE"
    className="App-tabContent"
    id="tabPane-PROFILE"
    role="tabpanel"
  >
    <div
      className="HorizontalGutter-root HorizontalGutter-double"
    >
      <div
        className="Flex-root"
      >
        <div
          className="Flex-flex Flex-halfItemGutter Flex-wrap"
        >
          <div
            className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
          >
            Signed in as 
            <span
              className="Typography-root Typography-bodyCopyBold Typography-colorTextPrimary"
            >
              Markus
            </span>
            .
          </div>
          <div
            className="Flex-root Typography-root Typography-bodyCopy Typography-colorTextPrimary Flex-flex"
          >
            <span>
              Not you? 
            </span>
            <button
              className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              type="button"
            >
              Sign Out
            </button>
          </div>
        </div>
      </div>
      <div
        className="HorizontalGutter-root HorizontalGutter-double"
      >
        <h1
          className="Typography-root Typography-heading3 Typography-colorTextPrimary"
        >
          Comment History
        </h1>
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        >
          <h1
            className="Typography-root Typography-heading4 Typography-colorTextPrimary"
          >
            Story: ⁨title⁩
          </h1>
          <time
            className="Timestamp-root RelativeTime-root"
            dateTime="2018-07-06T18:24:00.000Z"
            title="2018-07-06T18:24:00.000Z"
          >
            2018-07-06T18:24:00.000Z
          </time>
          <div
            className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
          >
            <div
              className="HTMLContent-root"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Joining Too",
                }
              }
            />
          </div>
          <div
            className="Flex-root Flex-flex Flex-itemGutter Flex-alignCenter Flex-directionRow"
          >
            <a
              className="BaseButton-root Button-root Button-sizeRegular Button-colorRegular Button-variantUnderlined"
              href="http://localhost/stories/story-1?commentID=comment-0"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              target="_parent"
              type="button"
            >
              <span
                aria-hidden="true"
                className="Icon-root Icon-sm"
              >
                launch
              </span>
              <span>
                View Conversation
              </span>
            </a>
          </div>
        </div>
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        >
          <h1
            className="Typography-root Typography-heading4 Typography-colorTextPrimary"
          >
            Story: ⁨title⁩
          </h1>
          <time
            className="Timestamp-root RelativeTime-root"
            dateTime="2018-07-06T18:24:00.000Z"
            title="2018-07-06T18:24:00.000Z"
          >
            2018-07-06T18:24:00.000Z
          </time>
          <div
            className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
          >
            <div
              className="HTMLContent-root"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "What's up?",
                }
              }
            />
          </div>
          <div
            className="Flex-root Flex-flex Flex-itemGutter Flex-alignCenter Flex-directionRow"
          >
            <a
              className="BaseButton-root Button-root Button-sizeRegular Button-colorRegular Button-variantUnderlined"
              href="http://localhost/stories/story-1?commentID=comment-1"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              target="_parent"
              type="button"
            >
              <span
                aria-hidden="true"
                className="Icon-root Icon-sm"
              >
                launch
              </span>
              <span>
                View Conversation
              </span>
            </a>
          </div>
        </div>
        <button
          aria-controls="talk-profile-commentHistory-log"
          className="BaseButton-root Button-root Button-sizeRegular Button-colorRegular Button-variantOutlined Button-fullWidth"
          disabled={false}
          id="talk-profile-commentHistory-loadMore"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          type="button"
        >
          Load More
        </button>
      </div>
    </div>
  </section>
</div>
`;
